src/components/plots/plot.ts

```diff
@@ -179,22 +183,33 @@ module Plottable {
      */
     public project(attrToSet: string, accessor: any, scale?: Scale<any, any>) {
       attrToSet = attrToSet.toLowerCase();
-      var currentProjection = this._projections[attrToSet];
-      var existingScale = currentProjection && currentProjection.scale;
+      var previousProjection = this._projections[attrToSet];
+      var previousScale = previousProjection && previousProjection.scale;
 
-      if (existingScale) {
-        this._datasetKeysInOrder.forEach((key) => {
-          existingScale._removeExtent(this.getID().toString() + "_" + key, attrToSet);
-          existingScale.broadcaster.deregisterListener(this);
-        });
+      accessor = Utils.Methods.accessorize(accessor);
+      this._projections[attrToSet] = {accessor: accessor, scale: scale, attribute: attrToSet};
+      this._updateExtentsForAttr(attrToSet);
+
+      if (previousScale) {
+        var listeningTopreviousScale = false;
```

Comment:
So let's turn the object into an array.

```typescript
var listeningToPreviousScale = Object.keys(this._projections).some((attr) => this._projections[attr].scale === previousScale);
```

On another note... javascript (or we) needs a `.values()` endpoint